feat: Build Stats

MaxIsJoe opened the pull request:
![image](https://github.com/user-attachments/assets/ecf22d9a-1cef-4ce1-ad46-4a393200ff38)

Adds a stats section at the top of the page that showcases how many additions, fixes and improvements were made in the past few builds + a card next to it that displays the latest code-scan version.

corp-0 committed:
chore: upgrade to nextjs15

MaxIsJoe committed:
Add download section to changelogs

corp-0 committed:
feat: Button component now supports choosing uppercase and side icons.

corp-0 committed:
fix: stupidly fix changelog page not loading enough builds to create the scrollbar

This will surely still break if someone has a comically tall monitor. That's trouble for another time tho

corp-0 committed:
refactor: revise Max's changes for new download dropdown

Eventually we have to make this component a general component that can be used anywhere else in the site

corp-0 committed:
feat: close dropdown after clicking on an element

semantic-release-bot committed:
docs(changelog): 1.12.0 [skip ci]

# [1.12.0](https://github.com/unitystation/unitystation-web/compare/v1.11.0...v1.12.0) (2024-10-27)

### Bug Fixes

* stupidly fix changelog page not loading enough builds to create the scrollbar ([a871f56](https://github.com/unitystation/unitystation-web/commit/a871f5654d7b010b1e18fb824ced75ac0b46cd88))

### Features

* Button component now supports choosing uppercase and side icons. ([1151eab](https://github.com/unitystation/unitystation-web/commit/1151eabdacad7dfdcfe859d26d359c16840eba5d))
* close dropdown after clicking on an element ([bf4e98e](https://github.com/unitystation/unitystation-web/commit/bf4e98e90decc286f7308c948b4a6cfd48631bea))

corp-0 committed:
chore: updates packages

corp-0 committed:
feat: changes "Weekly Update" pill's text to "Progress update"

We haven't been weekly for a while now

corp-0 committed:
chore: update packages

semantic-release-bot committed:
docs(changelog): 1.13.0 [skip ci]

# [1.13.0](https://github.com/unitystation/unitystation-web/compare/v1.12.0...v1.13.0) (2025-03-30)

### Features

* changes "Weekly Update" pill's text to "Progress update" ([6845ea6](https://github.com/unitystation/unitystation-web/commit/6845ea6b17fe2701e8fbdc4cecf64d1a1319ec97))

corp-0 committed:
feat: first iteration of the new ledger page

corp-0 committed:
feat: first iteration of the new ledger page

semantic-release-bot committed:
docs(changelog): 1.13.0 [skip ci]

# [1.13.0](https://github.com/unitystation/unitystation-web/compare/v1.12.0...v1.13.0) (2025-04-21)

### Features

* changes "Weekly Update" pill's text to "Progress update" ([6845ea6](https://github.com/unitystation/unitystation-web/commit/6845ea6b17fe2701e8fbdc4cecf64d1a1319ec97))
* first iteration of the new ledger page ([1cae21d](https://github.com/unitystation/unitystation-web/commit/1cae21d0368529be3fba38b3e8990ff6c61639a4))
* first iteration of the new ledger page ([cd5c42f](https://github.com/unitystation/unitystation-web/commit/cd5c42f3b569956608fcc80886f1b962d1ed1319))

MaxIsJoe committed:
feat: Build stats

MaxIsJoe committed:
feat: Build stats

corp-0 commented:
move your stats bellow the heading of the page. I will check the components and stuff after work

corp-0 commented on app/blog/[slug]/post.tsx:
```diff
@@ -14,7 +14,7 @@ const colourByType: Dict<string> = {
 const textByType: Dict<string> = {
     'announcement': 'Announcement',
-    'weekly': 'Weekly Update',
+    'weekly': 'Progress Update',
```
I think you forgot to rebase. This commmit already exists

corp-0 commented on app/changelog/buildStatsComponent.tsx:
```diff
@@ -0,0 +1,68 @@
+}
+
+const BuildStatsComponent: React.FC<BuildStatsProps> = ({ builds }) => {
+  const recentBuilds = builds.slice(0, 150);
```
150 is a big slice, isn't it?

corp-0 commented on app/changelog/buildStatsComponent.tsx:
```diff
@@ -0,0 +1,68 @@
+import React from 'react';
```
all components should be inside the outer components folder.

I'd say this component is an organism, so place it inside `components/organisms/changelog/`

corp-0 commented on components/mocules/Card.tsx:
```diff
@@ -0,0 +1,15 @@
+import React, { ReactNode } from 'react';
```
mocules?

corp-0 committed:
fix: updated vulnerable deps

corp-0 committed:
fix: updated vulnerable deps

corp-0 committed:
fix: updated vulnerable deps

corp-0 committed:
fix: update all deps

corp-0 committed:
fix: update all deps

semantic-release-bot committed:
docs(changelog): 1.13.0 [skip ci]

# [1.13.0](https://github.com/unitystation/unitystation-web/compare/v1.12.0...v1.13.0) (2025-12-10)

### Bug Fixes

* update all deps ([f307964](https://github.com/unitystation/unitystation-web/commit/f307964ce27f43656b783ece0db9e1d4f967eab0))
* updated vulnerable deps ([5826ba7](https://github.com/unitystation/unitystation-web/commit/5826ba71cfb41b1b5023b46f971b797119de6877))

### Features

* changes "Weekly Update" pill's text to "Progress update" ([6845ea6](https://github.com/unitystation/unitystation-web/commit/6845ea6b17fe2701e8fbdc4cecf64d1a1319ec97))
* first iteration of the new ledger page ([1cae21d](https://github.com/unitystation/unitystation-web/commit/1cae21d0368529be3fba38b3e8990ff6c61639a4))
* first iteration of the new ledger page ([cd5c42f](https://github.com/unitystation/unitystation-web/commit/cd5c42f3b569956608fcc80886f1b962d1ed1319))

semantic-release-bot committed:
docs(changelog): 1.13.0 [skip ci]

# [1.13.0](https://github.com/unitystation/unitystation-web/compare/v1.12.0...v1.13.0) (2025-12-10)

### Bug Fixes

* update all deps ([f307964](https://github.com/unitystation/unitystation-web/commit/f307964ce27f43656b783ece0db9e1d4f967eab0))
* updated vulnerable deps ([5826ba7](https://github.com/unitystation/unitystation-web/commit/5826ba71cfb41b1b5023b46f971b797119de6877))

### Features

* changes "Weekly Update" pill's text to "Progress update" ([6845ea6](https://github.com/unitystation/unitystation-web/commit/6845ea6b17fe2701e8fbdc4cecf64d1a1319ec97))
* first iteration of the new ledger page ([1cae21d](https://github.com/unitystation/unitystation-web/commit/1cae21d0368529be3fba38b3e8990ff6c61639a4))
* first iteration of the new ledger page ([cd5c42f](https://github.com/unitystation/unitystation-web/commit/cd5c42f3b569956608fcc80886f1b962d1ed1319))

MaxIsJoe committed:
Merge remote-tracking branch 'upstream/develop' into changelog-goodies

MaxIsJoe committed:
Merge branch 'changelog-goodies' of https://github.com/MaxIsJoe/unitystation-web into changelog-goodies
